src/components/shared/Widget/PriceChart.vue

```diff
@@ -791,25 +796,42 @@ export default class PriceChartWidget extends Mixins(
     const lastTimestamp = last(this.dataset)?.timestamp ?? Date.now();
 
     this.priceUpdateRequestId = requestId;
-
     await this.withApi(async () => {
       try {
-        const snapshots = await Promise.all(addresses.map((address) => this.fetchData(address)));
+        const snapshots: Snapshot[] = this.loadDataWhenReverse
+          ? addresses.map(
+              (address) => this?.snapshotBuffer?.[address] || { nodes: [], hasNextPage: false, endCursor: undefined }
+            )
+          : await Promise.all(addresses.map((address) => this.fetchData(address)));
 
-        // if no response, or tokens were changed, return
         if (!(snapshots && isEqual(addresses)(this.entities) && isEqual(requestId)(this.priceUpdateRequestId))) return;
-
         const pageInfos: Record<string, Partial<PageInfo>> = {};
         const dataset: SnapshotItem[] = [];
         const groups: SnapshotItem[][] = [];
         const timestamp =
           lastTimestamp ??
           Math.max(snapshots[0]?.nodes[0]?.timestamp ?? 0, snapshots[1]?.nodes[0]?.timestamp ?? 0) * 1000;
-
-        snapshots.forEach(({ hasNextPage, endCursor, nodes }, index) => {
+        snapshots.forEach(async ({ hasNextPage, endCursor, nodes }, index) => {
           const address = addresses[index];
           const buffer = this.samplesBuffer[address] ?? [];
           const normalized = normalizeSnapshots(buffer.concat(nodes), this.timeDifference, timestamp);
+
+          if (!this.snapshotBuffer[address]) {
+            this.snapshotBuffer[address] = {
+              nodes: [],
+              hasNextPage: false,
+              endCursor: undefined,
+            };
+          }
+
+          const existingNodes = this.snapshotBuffer[address]?.nodes;
```

Comment:
`this.snapshotBuffer[address].nodes` without optional chaining, because of `snapshotBuffer `initialization above